### PR TITLE
feat(ethereum): add support for EIP-7702

### PR DESCRIPTION
**Pull request by fubuloubu**
### What I did

Added support to decode EIP-7702 `SET_CODE` transactions (tx type 0x4)

### How I did it

shout out to Web3.py

### How to verify it

```python
$ ape console --network :mainnet:node
INFO:     Connecting to a 'mevblocker' node.

In [1]: tx = chain.get_receipt("0x00645e1eafe945e7a356ccd2be36dcef81eb3bb752035887487cf82b6cc13c28")

In [2]: tx.transaction.authorizations[0].authority
Out[2]: '0x864a5a451126D99647D9fE58b21e124AE6F67E35'
```

### Checklist

- [x] All changes are completed
- [x] Change is covered in tests
~~- [ ] Documentation is complete~~

### CODE REVIEW

File: src/ape_ethereum/transactions.py [hunk 1]
```diff
@@ -186,6 +189,41 @@ class SharedBlobTransaction(DynamicFeeTransaction):
...
 
+class Authorization(BaseModel):
+    """
+    `EIP-7702 <https://eips.ethereum.org/EIPS/eip-7702>`__ authorization list item.
+    """
+
+    chain_id: HexInt = Field(alias="chainId")
+    address: AddressType
+    nonce: HexInt
+    v: HexInt = Field(alias="yParity")
```
Comment (antazoey) [inline src/ape_ethereum/transactions.py:200]: Option: we could have a model validator make a message signature and model serializer serialize back to v,r,s in the JSON

**Comment by fubuloubu** [comment]
@antazoey could use the help here, the test failure is... perplexing

Locally it is passing for me

**Comment by fubuloubu** [comment]
> @antazoey could use the help here, the test failure is... perplexing
> 
> Locally it is passing for me

Tracking this error in #2619, I think I solved it by down-pinning Click to <8.2

File: src/ape_ethereum/transactions.py (hunk 1 above)
Comment (fubuloubu) [reply src/ape_ethereum/transactions.py:200]: I was considering this, but to be honest I would rather have the constituent values, and then compute the signature on the fly when needed

**Comment by fubuloubu** [comment]
Still getting some malformed errors when submitting these txs through public RPCs/with foundry

eth-tester doesn't support it yet

**Review by antazoey — APPROVED** [review]
You can merge, tests are fine. I don't know why CI/CD is being weird -- Geth has been unusual lately. We can fix the tests on a separate branch.

**Comment by fubuloubu** [comment]
> You can merge, tests are fine. I don't know why CI/CD is being weird -- Geth has been unusual lately. We can fix the tests on a separate branch.

Once I successfully submit a type 4 txn w/ #2613 I will merge this one. Keep finding corner cases there that might need to update this PR
